configs/test/bot/setup/android.bash

```diff
@@ -0,0 +1,97 @@
+  exit 1
+fi
+
+if [ -z "$GOOGLE_APPLICATION_CREDENTIALS" ]; then
+  echo "\$GOOGLE_APPLICATION_CREDENTIALS is not set."
+  exit 1
+fi
+
+if [ -z "$ANDROID_SERIAL" ]; then
+  echo "\$ANDROID_SERIAL is not set."
+  exit 1
+fi
+
+NFS_ROOT=  # Fill in NFS information if available.
+APPENGINE=google_appengine
+APPENGINE_FILE=google_appengine_1.9.75.zip
+GOOGLE_CLOUD_SDK=google-cloud-sdk
+GOOGLE_CLOUD_SDK_ARCHIVE=google-cloud-sdk-232.0.0-linux-x86_64.tar.gz
+INSTALL_DIRECTORY=${INSTALL_DIRECTORY:-${HOME}}
+APPENGINE_DIR="$INSTALL_DIRECTORY/$APPENGINE"
+DEPLOYMENT_BUCKET="deployment.$CLOUD_PROJECT_ID.appspot.com"
+GSUTIL_PATH="$INSTALL_DIRECTORY/$GOOGLE_CLOUD_SDK/bin"
+ROOT_DIR="$INSTALL_DIRECTORY/clusterfuzz"
+PYTHONPATH="$PYTHONPATH:$APPENGINE_DIR:$ROOT_DIR/src"
+ADB_PATH="$INSTALL_DIRECTORY/platform-tools"
+PATH="$PATH:$ADB_PATH"
+
+echo "Creating directory $INSTALL_DIRECTORY."
+if [ ! -d "$INSTALL_DIRECTORY" ]; then
+  mkdir -p "$INSTALL_DIRECTORY"
+fi
+
+cd $INSTALL_DIRECTORY
+
+echo "Fetching Google Cloud SDK."
+if [ ! -d "$INSTALL_DIRECTORY/$GOOGLE_CLOUD_SDK" ]; then
+  curl -O "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/$GOOGLE_CLOUD_SDK_ARCHIVE"
+  tar -xzf $GOOGLE_CLOUD_SDK_ARCHIVE
+  rm $GOOGLE_CLOUD_SDK_ARCHIVE
+fi
+
+echo "Fetching Android platform tools for ADB."
+if [ ! -d "$ADB_PATH" ]; then
+  wget https://dl.google.com/android/repository/platform-tools-latest-linux.zip
+  unzip $INSTALL_DIRECTORY/platform-tools-latest-linux.zip
+  rm $INSTALL_DIRECTORY/platform-tools-latest-linux.zip
+fi
+
+echo "Fetching Google App Engine SDK."
+if [ ! -d "$INSTALL_DIRECTORY/$APPENGINE" ]; then
+  curl -O "https://commondatastorage.googleapis.com/clusterfuzz-data/$APPENGINE_FILE"
+  unzip -q $APPENGINE_FILE
+  rm $APPENGINE_FILE
+fi
+
+echo "Ensure device is connected."
+if ! $ADB_PATH/adb devices -l | grep -q "$ANDROID_SERIAL"; then
+  echo "Device $ANDROID_SERIAL is not connected."
+  exit 1
+fi
+
+# Otherwise, gsutil will error out due to multiple types of configured
+# credentials. For more information about this, see
+# https://cloud.google.com/storage/docs/gsutil/commands/config#configuration-file-selection-procedure
+echo "Specifying the proper Boto configuration file."
+BOTO_CONFIG_PATH=$($GSUTIL_PATH/gsutil -D 2>&1 | grep "config_file_list" | egrep -o "/[^']+gserviceaccount\.com/\.boto") || true
+if [ -f $BOTO_CONFIG_PATH ]; then
+  export BOTO_CONFIG="$BOTO_CONFIG_PATH"
+else
+  echo "WARNING: failed to identify the Boto configuration file and specify BOTO_CONFIG env."
+fi
+
+echo "Downloading ClusterFuzz source code."
+$GSUTIL_PATH/gsutil cp gs://$DEPLOYMENT_BUCKET/linux.zip clusterfuzz-source.zip
+unzip -f -q clusterfuzz-source.zip
+
+echo "Running ClusterFuzz."
+OS_OVERRIDE="ANDROID" PATH="$PATH" NFS_ROOT="$NFS_ROOT" GOOGLE_APPLICATION_CREDENTIALS="$GOOGLE_APPLICATION_CREDENTIALS" ROOT_DIR="$ROOT_DIR" PYTHONPATH="$PYTHONPATH" GSUTIL_PATH="$GSUTIL_PATH" python $ROOT_DIR/src/python/bot/startup/run.py &
```

Comment:
Needs to pass ANDROID_SERIAL as well.

Reply:
We assume this is already exported. It should be fine to include it since that matches some of the other invocations, though.